wrtbwmon

```diff
@@ -192,22 +198,27 @@ update()
     checkDB
     checkWAN
 
+    lock
+
     > /tmp/ipv6_neighbours_$$.tmp
     if [ $ipv6 -eq 1 ]; then
-        ip -6 neighbour show > /tmp/ipv6_neighbours_$$.tmp 2>/dev/null
+        $ip -6 neighbour show > /tmp/ipv6_neighbours_$$.tmp 2>/dev/null
     fi
+
     > /tmp/iptables_$$.tmp
-    lock
```

Comment:
Why was this lock removed? We need the lock to protect both the iptables accounting rules and the DB file.

Reply:
It's (literally) been years since I've looked at this code - so I can't offer anything terrible useful - but it looks like I moved the lock up (line 201).